client/network/src/behaviour.rs

```diff
@@ -307,18 +344,41 @@ Behaviour<B, H> {
 	}
 }
 
+impl<B: BlockT, H: ExHashT> NetworkBehaviourEventProcess<request_responses::Event> for Behaviour<B, H> {
+	fn inject_event(&mut self, event: request_responses::Event) {
+		match event {
+			request_responses::Event::InboundRequest { peer, protocol, outcome } => {
+				self.events.push_back(BehaviourOut::InboundRequest {
+					peer,
+					protocol,
+					outcome,
+				});
+			}
+
+			request_responses::Event::OutboundFinished { request_id, outcome } => {
+				self.events.push_back(BehaviourOut::RequestFinished {
+					request_id,
+					outcome,
+				});
+			},
+		}
+	}
+}
+
 impl<B: BlockT, H: ExHashT> NetworkBehaviourEventProcess<block_requests::Event<B>> for Behaviour<B, H> {
 	fn inject_event(&mut self, event: block_requests::Event<B>) {
 		match event {
 			block_requests::Event::AnsweredRequest { peer, total_handling_time } => {
-				self.events.push_back(BehaviourOut::AnsweredRequest {
+				let protocol = maybe_utf8_bytes_to_string(&self.block_requests.protocol_name())
```

Comment:
The protocol name contained in `BlockRequests` unfortunately contains the `ProtocolId`, which might theoretically not be UTF-8. In practice, this `ProtocolId` is always UTF-8, and we add the requirement that the `ProtocolId` must be UTF-8, but this was a bit too off-topic for this PR.

Reply:
Ideally this would be handled consistently. If changing `BlockRequests::protocol_name` to return a `&str` is too much for this PR maybe protocol names should continue to be `&[u8]` for the time being and a follow-up PR could transition to `&str` everywhere at once.

Reply:
I understand that you hate `maybe_utf8_bytes_to_string`, but this is not something that this PR introduces, and the change is out of scope to me.